File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/SshGenerateRequest.java
package org.cloudfoundry.credhub.requests;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

public class SshGenerateRequest extends BaseCredentialGenerateRequest {
  @JsonProperty("parameters")
  private SshGenerationParameters generationParameters;

  @Override
  @JsonIgnore
  public GenerationParameters getGenerationParameters() {
    if (generationParameters == null) {
      generationParameters = new SshGenerationParameters();
    }
    return generationParameters;
  }

  public void setGenerationParameters(final SshGenerationParameters generationParameters) {
    this.generationParameters = generationParameters;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/SshGenerateRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

class SshGenerateRequest : BaseCredentialGenerateRequest() {
    @JsonProperty("parameters")
    override var generationParameters: SshGenerationParameters? = null
        @JsonIgnore
        get() {
            if (field == null) {
                field = SshGenerationParameters()
            }
            return field
        }
}
